conans/client/rest/rest_client_v1.py

```diff
@@ -117,12 +118,23 @@ def _get_file_to_url_dict(self, url, data=None):
         urls = self.get_json(url, data=data)
         return {filepath: complete_url(self.remote_url, url) for filepath, url in urls.items()}
 
+    def _add_matrix_params(self, urls):
+        if not self._matrix_params:
+            return urls
+
+        matrix_str = self.router._matrix_params_str
```

Comment:
This "private" access to the router is a bit weird. Should this belong to the router? Not sure but something a bit weird.